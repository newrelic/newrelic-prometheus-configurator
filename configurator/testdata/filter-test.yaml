kubernetes:
  jobs:
    - job_name_prefix: kubernetes-job
      target_discovery:
        pod: true
        endpoints: true
        # cannot add more labels/annotations to this test since the order is not
        # guaranteed when checking expected 'regex' results. `kubernetes_test.go` for more
        # complex tests.
        filter:
          annotations:
            prometheus.io/scrape: true
          
newrelic_remote_write:
  license_key: nrLicenseKey
