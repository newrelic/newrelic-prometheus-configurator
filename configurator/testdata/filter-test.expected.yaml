scrape_configs:
  - job_name: kubernetes-job-pod
    relabel_configs:
    # Filter builder configs
      - source_labels: ["__meta_kubernetes_pod_annotation_prometheus_io_scrape"]
        regex: "true"
        separator: ";"
        action: keep
    # TODO pod configs will be needed when the feature is added
        
  - job_name: kubernetes-job-endpoints
    honor_labels: true
    kubernetes_sd_configs:
      - role: endpoints
    relabel_configs:
    # Filter builder configs
      - source_labels: ["__meta_kubernetes_service_annotation_prometheus_io_scrape"]
        regex: "true"
        separator: ";"
        action: keep
    # Endpoints builder configs
      - source_labels: [ __meta_kubernetes_pod_phase ]
        action: drop
        regex: Pending|Succeeded|Failed|Completed
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scheme]
        action: replace
        target_label: __scheme__
        regex: (https?)
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [ __address__, __meta_kubernetes_service_annotation_prometheus_io_port ]
        action: replace
        regex: (.+?)(?::\d+)?;(\d+)
        target_label: __address__
        replacement: $1:$2
      - action: labelmap
        regex: __meta_kubernetes_service_annotation_prometheus_io_param_(.+)
        replacement: __param_$1
      - action: labelmap
        regex: __meta_kubernetes_service_label_(.+)
      - source_labels: [__meta_kubernetes_namespace]
        action: replace
        target_label: namespace
      - source_labels: [__meta_kubernetes_service_name]
        action: replace
        target_label: service
      - source_labels: [ __meta_kubernetes_pod_node_name ]
        action: replace
        target_label: node

remote_write:
  - url: https://metric-api.newrelic.com/prometheus/v1/write
    authorization:
      credentials: nrLicenseKey
